x/emissions/keeper/block_rewards_components.go

```diff
@@ -18,9 +18,14 @@ func (k Keeper) GetBlockRewardComponents(ctx sdk.Context) (sdk.Dec, sdk.Dec, sdk
 }
 
 func (k Keeper) GetBondFactor(ctx sdk.Context, stakingKeeper types.StakingKeeper) sdk.Dec {
-	targetBondRatio := sdk.MustNewDecFromStr(k.GetParamsIfExists(ctx).TargetBondRatio)
-	maxBondFactor := sdk.MustNewDecFromStr(k.GetParamsIfExists(ctx).MaxBondFactor)
-	minBondFactor := sdk.MustNewDecFromStr(k.GetParamsIfExists(ctx).MinBondFactor)
+	params, found := k.GetParams(ctx)
+	if !found {
+		// TODO: what to return here?
```

Comment:
This is same as above comment (and for other situation where this function is used)

I suggest giving the params as argument, so we just handle params not existing at the top level

Also since reading the state would no longer be necessary, we could move this logic in the `types` packages and makes the code more functional paradigm